dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/StatsTaskListGenerator.java

```diff
@@ -16,17 +16,64 @@
  */
 package com.google.edwmigration.dumper.application.dumper.connector.oracle;
 
+import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
 import com.google.edwmigration.dumper.application.dumper.ConnectorArguments;
+import com.google.edwmigration.dumper.application.dumper.task.JdbcSelectTask;
 import com.google.edwmigration.dumper.application.dumper.task.Task;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
 @ParametersAreNonnullByDefault
 class StatsTaskListGenerator {
 
+  private static final ImmutableList<StatsQuery> QUERIES =
+      ImmutableList.of(
+          // TODO: add entries for other SQLs to this list
+          StatsQuery.create("hist-cmd-types", Tool.STATSPACK));
+
   @Nonnull
-  ImmutableList<Task<?>> createTasks(ConnectorArguments arguments) {
-    throw new UnsupportedOperationException();
+  ImmutableList<Task<?>> createTasks(ConnectorArguments arguments) throws IOException {
+    ImmutableList.Builder<Task<?>> builder = ImmutableList.<Task<?>>builder();
+
+    for (StatsQuery item : QUERIES) {
+      builder.add(item.toTask());
+    }
+    return builder.build();
+  }
+
+  enum Tool {
+    AWR("awr"),
```

Comment:
User should have an option to choose between AWR and STATSPACK queries depending on what tool is used by the customer. This can be ofc addressed later.

Reply:
Thanks, I'll add it later.

I think if it's going to be at most one value for `Tool` per execution, then we can use this information and change things slightly.
The directory could be dropped from the output paths and instead we could use the metadata YAML to inform about which tool was used.

Reply:
> The statspack directory could be dropped from the output paths and instead we could use the metadata YAML to inform about which tool was used

Thanks, I think this would be a great solution. Metadata YAML already includes information about dumper parameters so we can use it.